release/plugins/conf.py

```diff
@@ -1,12 +1,49 @@
 import argparse
+from dataclasses import MISSING, dataclass, fields
 from enum import Enum
-from pathlib import Path
-from typing import NamedTuple
 
 import yaml
 from semver import VersionInfo, bump_minor, bump_patch
 
 from .common import Hooks
+from .env import ENV_VARIABLE_NAMES, get_parameter
+
+
+class ParameterMixin:
+    SUBCLASSES = []
+
+    def __init_subclass__(cls, **kwargs):
+        # collect subclasses for validation
+        # validation can't be done here as dataclass decorator
+        # is not executed at this point
+        cls.SUBCLASSES.append(cls)
+
+    @classmethod
+    def validate(cls):
+        """Validate that all parameters have env variable representation"""
+        for subclass in cls.SUBCLASSES:
+            env_variable_names = ENV_VARIABLE_NAMES.get(subclass.__name__, {})
```

Comment:
Why not just run this check in `__init_subclass__`, without tracking SUBCLASSES?

Reply:
https://github.com/TangleInc/release-tool/blob/4ab6b4e5bb12c910c792af19cf2662b91a1e80e1/release/plugins/conf.py#L15:L19